File: ddd-infrastructure/src/main/java/com/zhuzichu/infrastructure/converter/UserConverter.java
package com.zhuzichu.infrastructure.converter;

import com.zhuzichu.domain.entity.User;
import com.zhuzichu.infrastructure.dto.LoginDto;

import org.mapstruct.Mapper;
import org.mapstruct.factory.Mappers;

@Mapper
public interface UserConverter {

    UserConverter INSTANCE = Mappers.getMapper(UserConverter.class);

    User toUser(LoginDto dto);

}

File: ddd-infrastructure/src/main/java/com/zhuzichu/infrastructure/converter/UserConverter.kt
package com.zhuzichu.infrastructure.converter

import com.zhuzichu.infrastructure.dto.LoginDto
import com.zhuzichu.domain.entity.User
import com.zhuzichu.shared.tool.object2Object
import javax.inject.Inject

class UserConverter @Inject constructor() {

    fun toUser(loginDto: LoginDto?): User {
        val user = object2Object(loginDto, User::class.java)
        return user!!
    }

}